packages/job-worker/src/cache/CacheCollection.ts

```diff
@@ -116,11 +81,20 @@ export class DbCacheReadCollection<TDoc extends { _id: ProtectedString<any> }> {
 		if (span) span.end()
 		return res
 	}
-	findOne(
-		selector: MongoQuery<TDoc> | TDoc['_id'] | SelectorFunction<TDoc>,
-		options?: FindOneOptions<TDoc>
-	): TDoc | undefined {
-		return this.findFetch(selector, options)[0]
+	findOne(selector: TDoc['_id'] | SelectorFunction<TDoc>, options?: FindOneOptions<TDoc>): TDoc | undefined {
```

Comment:
Selector should probably allow `null` here as well, or is this intentional?

Reply:
This was intentional. `null` would pick the 'first' document in the map, which would be a rather random and potentially unstable result. Outside of unit tests I dont see it as having any value